app/helpers/i18n_helper.rb

```diff
@@ -47,6 +47,39 @@ def date_eras_gregorian(date)
     nil
   end
 
+  # Return the most accurate date from an array of dates using scoring algorithm below, if a single
+  # date string is passed, then that value is returned immediately.
+  def date_most_accurate(dates)
+    return nil unless dates.present?
+    if dates.is_a?(String)
+      return date_eras_gregorian(dates) || dates.strip
+    end
```

Comment:
if we're keeping this in here, I think the above 4 lines could be simplified and prevent additional processing of empty arrays by changing them to:
```
 return nil if dates.blank?
 return date_eras_gregorian(dates) || dates.strip if dates.is_a?(String)
```